Source/WTF/wtf/Assertions.cpp

```diff
@@ -700,6 +704,35 @@ void WTFCrashWithInfoImpl(int, const char*, const char*, int, uint64_t) { CRASH(
 
 #endif // (OS(DARWIN) || PLATFORM(PLAYSTATION)) && (CPU(X64_64) || CPU(ARM64))
 
+#if PLATFORM(COCOA)
+void WTFPGMCrashDueToOOB()
+{
+    pas_probabilistic_guard_malloc_initialize_pgm_for_testing();
+
+    /* PGM is only enabled on large allocations, so we need to set a large allocation size. */
+    void* result = bmalloc_allocate(100000000, pas_non_compact_allocation_mode);
+    char* lowerGuardPage = static_cast<char*>(result) - getpagesize();
+    *lowerGuardPage = 'a';
+
+    __builtin_unreachable();
+}
```

Comment:
Would also be great to create both an OOB overflow and OOB underflow test functions since detection is different for each one.

Reply:
Added support for under and over flow tests.